File: sso/src/main/java/com/web/sso/model/ResBody.java
package com.web.sso.model;

public class ResBody<T> {
    private String status_code;

    private String message;

    private T data;

    public ResBody() {}

    public ResBody(String status_code, String message, T data) {
        this.status_code = status_code;
        this.message = message;
        this.data = data;
    }

    public String getStatuscode() {
        return status_code;
    }

    public void setStatuscode(String status_code) {
        this.status_code = status_code;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }
}

File: myapp/lib/model/ResBody.dart
class ResBody {
  String message;
  String statuscode;
  String data;

  ResBody({
    this.message,
    this.statuscode,
    this.data,
  });

  factory ResBody.fromJson(Map<String, dynamic> json) {
    return ResBody(message: json['message'], statuscode: json['statuscode'], data: json['data']);
  }
}